R/utils-get_code_dependency.R

```diff
@@ -35,17 +35,22 @@ get_code_dependency <- function(code, names, check_names = TRUE) {
 
   code <- parse(text = code, keep.source = TRUE)
   pd <- utils::getParseData(code)
+  calls_pd <- extract_calls(pd)
 
   if (check_names) {
     # Detect if names are actually in code.
-    symbols <- unique(pd[pd$token == "SYMBOL", "text"])
-    if (!all(names %in% symbols)) {
+    symbols <- unlist(lapply(calls_pd, function(call) call[call$token == "SYMBOL", "text"]))
+    if (any(pd$text == "assign")) {
+      assign_calls <- Filter(function(call) any(call$token == "SYMBOL_FUNCTION_CALL" & call$text == "assign"), calls_pd)
+      ass_str <- unlist(lapply(assign_calls, function(call) call[call$token == "STR_CONST", "text"]))
+      ass_str <- gsub("'", "", ass_str)
```

Comment:
What about regular (escaped) quotation marks? How about `gsub("['\"]", "", ass_str)` or `gsub("^['\"]|['\"]$", "", ass_str)`?

Reply:
will double check, thanks :)!

Reply:
Just pushed a fix for that, and also extended the mentioned test ff6c2fd37ef302e56a42a7950c3dad7b1d4f909d